t/t3422-rebase-incompatible-options.sh

```diff
@@ -61,7 +61,6 @@ test_rebase_am_only () {
 }
 
 test_rebase_am_only --whitespace=fix
-test_rebase_am_only --ignore-whitespace
 test_rebase_am_only --committer-date-is-author-date
 test_rebase_am_only -C4
 
```

Comment:
add more incompatibility tests? Like `--rebase-merges` in incompatible with `--strategy` et al.

Reply:
Possibly, but that would also be the topic for another patch.  I didn't check if there already is some such test, but if we wanted to test that it would be out of scope for this patch series.

If we can however come up with a test for interactive rebase with `--ignore-whitespace`, that might be nice.

Reply:
Ah you mean also adding a test of `--rebase-merges` being incompatible with `--ignore-whitespace` for now?  That might possibly a good thing to do, though not sure it adds too much value.

Reply:
It will provide the additional benefit of increased test coverage. But like @dscho mentioned, some veterans not using the test suite (as it already takes a lot of time) and submitting the code, kinda seems worthless.

Reply:
There's nothing fundamentally incompatible between --ignore-whitespace and --rebase-merges.  Yes, it adds a strategy-option, but the only reason strategy-options are incompatible with --rebase-merges is because no one implemented translating strategy-options into the necessary -X flags in the todo file for rebase-merges...which strikes me more as laziness of implementation.  Adding a test that these two are incompatible suggests there's a real limitation and perhaps that we even want them to be known to be incompatible.  So, I'm not so sure this is that valuable.

Reply:
@newren @tgummerer remember that I posted [this](https://gist.github.com/r1walz/ca9ce5882a4d6afd1cb7e7c676ed835c) gist containing setup script? I am thinking of adding that as a test. If you agree, where should I add it?

Reply:
Something like that should work, though we'd want to test the merge backend of rebase and the --ignore-whitespace flag, not cherry-pick.

I'll turn that question around. Where do you think it should be added, and why there? :) If you have a hard time coming up with that though, I'm happy to give a suggestion and walk you through my thoughts though

Reply:
> Something like that should work, though we'd want to test the merge backend of rebase and the --ignore-whitespace flag, not cherry-pick.

Well ofc, I'll change it to match `rebase -i` :sweat_smile:.

> I'll turn that question around. Where do you think it should be added, and why there? :) 

`t3402`? Since it already contains tests related `--merge` backend.

Reply:
I usually go into the testsuite directory (t/) and do either something like 'git grep rebase.*ignore' or 'ls *rebase*' and look for places that already have similar tests.  I don't actually see one, so we could create a t3431-$SOMETHING.sh, or just pick another semi-reasonable file to add them into.  I'm actually partial to adding them to t3402-rebase-merge.sh, in part because it's a merge-related code change and in part because the old merge backend is dead (the interactive backend was extended to handle everything it could and thus replaced it) so we might as well give it a more useful purpose by adding new and useful tests to it.